file.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"fmt"
+	"net/http"
+	"io"
+	"path/filepath"
+	"bytes"
+	"archive/zip"
+	"strings"
+)
+
+// Download the zip file at the given URL to a temporary local directory.
+// Returns the asbolute path to the downloaded zip file.
+// IMPORTANT: You must call defer os.RemoveAll(dir) when done with this directory!
+func downloadGithubZipFile(repoOwner, repoName, gitTag, githubToken string) (string, *fetchError) {
+	// Create a temp directory
+	tempDir, err := ioutil.TempDir("", "")
```

Comment:
That's the catch with languages like Go. Speed of `writing` is increased. But speed of reading/maintaining is severely decreased. And as you do the latter approximately 10x as much as the former, that's usually a poor trade-off.